tests/unit-tests/unsupported-theme-handlers/test-class-sensei-unsupported-theme-handler-page-imitator.php

```diff
@@ -0,0 +1,172 @@
+		fclose( $handle );
+	}
+
+	/**
+	 * Delete the page template added for testing.
+	 */
+	public static function delete_page_template() {
+		// Remove the page.php template.
+		$filename = get_template_directory() . '/page.php';
+		unlink( $filename );
+	}
+
+	/**
+	 * Ensure Sensei_Unsupported_Theme_Handler_Faux_Page_Imitator render runs by checking action.
+	 *
+	 * @since 1.12.0
+	 */
+	public function testShouldRunTestActionInRender() {
+		$test_action = Sensei_Unsupported_Theme_Handler_Faux_Page_Imitator::TEST_ACTION;
+
+		// We'll test to ensure it uses the template by checking if the
+		// test action action was run.
+		$this->assertEquals(
+			0,
+			did_action( $test_action ),
+			'Should not have already done action ' . $test_action
+		);
+
+		$this->handler->handle_request();
+
+		$this->assertEquals(
+			1,
+			did_action( $test_action ),
+			'Should have done action ' . $test_action
+		);
+	}
+
+	/**
+	 * Ensure Sensei_Unsupported_Theme_Handler_Faux_Page_Imitator sets up a dummy post for
+	 * the final render of the faux content.
+	 *
+	 * @since 1.12.0
+	 */
+	public function testShouldSetupDummyPost() {
+		global $post;
+
+		$this->handler->handle_request();
+
+		$this->assertEquals( 0, $post->ID, 'The dummy post ID should be 0' );
+		$this->assertEquals( 'page', $post->post_type, 'The dummy post type should be "page"' );
+
+		$copied_from_course = array(
+			'post_author',
+			'post_date',
+			'post_date_gmt',
+			'post_modified',
+			'post_modified_gmt',
+		);
+		foreach ( $copied_from_course as $field ) {
+			$this->assertEquals(
+				$this->course->{$field},
+				$post->{$field},
+				"Dummy post field $field should be copied from the Course"
+			);
+		}
+	}
+
+	/**
+	 * Ensure Sensei_Unsupported_Theme_Handler_Faux_Page_Imitator renders the dummy post
+	 * with the 'page' template if it's available.
+	 *
+	 * @since 1.12.0
+	 */
+	public function testShouldRenderDummyPostWithPageTemplate() {
+		$this->handler->handle_request();
+		$template_file = apply_filters( 'template_include', 'a-random-template.php' );
+		$this->assertEquals( get_template_directory() . '/page.php', $template_file );
+	}
+
+	/**
+	 * Ensure Sensei_Unsupported_Theme_Handler_Faux_Page_Imitator hides the dummy post
+	 * title when it is rendered.
+	 *
+	 * @since 1.12.0
+	 */
+	public function testShouldHideDummyPostTitle() {
+		global $post;
+
+		$this->handler->handle_request();
+
+		$this->assertEquals(
+			'',
+			apply_filters( 'the_title', 'Dummy Post Title', $post->ID ),
+			'Title of the dummy post should be blank'
+		);
+		$this->assertEquals(
+			'Course Title',
+			apply_filters( 'the_title', 'Course Title', $this->course ),
+			'Title of the course post should no be blank'
+		);
+	}
+
+	/**
+	 * Helper to set up the current request to be a Module page. This request
+	 * will be handled by the unsupported theme handler if the theme is not
+	 * supported.
+	 *
+	 * @since 1.12.0
+	 */
+	private function setupUnsupportedPage() {
```

Comment:
The comment says that this is setting up a Module page. In the code, it looks like it's setting up a regular Post. But it's also creating a Course object. This is a bit confusing...maybe we could clarify this a bit?